.travis.yml

```diff
@@ -0,0 +1,26 @@
+sudo: false
+
+notifications:
+  email: false
+
+services:
+  - docker
+
+env:
+  matrix:
+    - PYTHON_VERSION=3.6
+    - PYTHON_VERSION=3.7
+
+addons:
+  apt:
+    packages:
+    - docker-ce
+
+matrix:
+  fast_finish: true
+
+before_install:
+  - docker build -t invenio ./python$PYTHON_VERSION/
```

Comment:
Perhaps you can use the full image name here:

```
docker build -t inveniosoftware/centos7-python ...
```

for the sake of consistency, even though the image won't be pushed by Travis.


BTW speaking of https://hub.docker.com/u/inveniosoftware/ I don't think `centos7-python` is a good base image name, because:

- based on the name alone, seeing `centos7-python:3.7`, people may think we are providing a rather general customised CentOS7 base image adding Python-3.7 only -- while the image does several more Invenio-specific things, such as installing `npm`, which people may not want;

- if we need to differentiate one day between `npm6` and `npwm8`, then how would be expressed in the image? via different name? via different tag?

- if we make mistake and need to repush several different versions of the Invenio 3.1.2 base image using Python 3.8, how would this be represented in the tag?  3.8.1-N where N is our local "build" version, so to speak?

- imagine that Invenio 3.2 or other such later version will change something in the way how npm is used, or how the instance directories are created; in this case, both Invenio 3.0 and Invenio 3.2 release series will be runnable on Python 3.7. How would the two base images be distinguishable?

I think a more future-proof naming schema would be something like:

```
invenio-centos7-base:3.0.0
invenio-centos7-base:3.0.0-py37
invenio-centos7-base:3.2.0-py37-npm6
invenio-centos7-base:3.2-py37-npm8
```

or other such alternative naming scheme such as `invenio3-centos7-py36:1.0.0` which would however lead to a proliferation of various image names instead of just tags.


This should better express that (a) the image is not general Python customisation but meant as a base image for Invenio instances, (b) that it is suitable for some particular Invenio release series version, and (c) that it customises more than py37 should a need arise one day to distinguish between npm6 and npm8, say.

Reply:
For the sake of further discussion about how to choose tags, it may be interesting to recall that we need to cover various upgrade scenarios: (1) Python version upgrades e.g. 3.7.0->3.7.1; (2) CentOS7 based image upgrades e.g. security; (3) Invenio dependency updates e.g. Invenio 3.0.x necessitating different directory structure or different npm package than Invenio 3.1.x; (4) incremental build number (or SHA1) for the image to cover for our own packaging/building errors.  

Basically, we are packaging Python-on-CentOS for the use of various Invenio installations that may be on various versions etc, so we are doing the work here that is very similar to what a typical GNU/Linux distribution packager does (e.g. `linux-headers 4.18.5.arch1-1`, `2.6.0-1ubuntu1`, etc).

Having well-though and extensible tag structure is important for any production service that may be based on these images and that deploys latest upgrades and that need to quickly roll back in case of unforeseen problems.

Examples to illustrate discussion:

```
FROM inveniosoftware/centos7-python:3.6
FROM inveniosoftware/centos7-python:3.7

FROM inveniosoftware/centos7-python:3.6-invenio30
FROM inveniosoftware/centos7-python:3.6-invenio30
FROM inveniosoftware/centos7-python:3.7-invenio32
FROM inveniosoftware/centos7-python:3.7-invenio32

FROM inveniosoftware/centos7-python:3.7.0-invenio30
FROM inveniosoftware/centos7-python:3.7.1-invenio30

FROM inveniosoftware/centos7-python:3.7.2-invenio30-1
FROM inveniosoftware/centos7-python:3.7.2-invenio30-2

FROM inveniosoftware/centos7-python:3.7.0-invenio30-3025579
FROM inveniosoftware/centos7-python:3.7.1-invenio30-26ccb52
```